numpy/lib/function_base.py

```diff
@@ -1912,10 +1911,16 @@ def diff(a, n=1, axis=-1):
     slice2[axis] = slice(None, -1)
     slice1 = tuple(slice1)
     slice2 = tuple(slice2)
+
+    if a.dtype == bool:
```

Comment:
Also it would be nice if you can make the commit use our typical format such as `BUG: ...` (or maybe `MAINT:` here even, sometimes its hard to say ;)).